gettsim/pre_processing/piecewise_functions.py

```diff
@@ -372,20 +386,14 @@ def piecewise_polynomial_float(
     if (x < lower_thresholds[0]) or (x > upper_thresholds[-1]) or np.isnan(x):
         return np.nan
     index_interval = np.searchsorted(upper_thresholds, x, side="left")
-    if rates_modified:
```

Comment:
Can you explain why you moved it into the function that is called in every iteration? Sort of defies the purpose of the whole thing. Maybe fine for now given time pressure, but we should open an issue to fix it then.

Reply:
We need to tell the function if it can use the recalculated intercepts! If we have individual rates we can not use the intercepts anymore and need to calculate everything from the beginning!

Reply:
I guess I do not understand the concept of individual rates. Please leave as is for now and make an issue to revisit.